feat(certificate viewer): create floating button bar

simboonlong opened the pull request:
feat: add basic transitions to buttons
feat: smaller share buttons when in mobile breakpoint

simboonlong committed:
feat(certificate viewer): create floating button bar

add a wrapper .share-buttons div
position fixed to .share-buttons div
smaller share buttons when in mobile breakpoint
add basic transitions
add media-breakpoint mixin

simboonlong committed:
lint fix

Nebulis commented on src/components/certificateViewer.scss:
```diff
@@ -143,6 +157,13 @@
     width: 100%;
   }
+
+  @include media-breakpoint-custom($max: 767px) {
```
why not just max-width ?

simboonlong replied:
used @media max-width instead. leaving the custom breakpoint mixin in variables.scss, in case if next time need to handle edge cases for in between min + max scenarios.

sebastianquek reviewed (commented):
I've got 3 comments regarding the sizing and alignment:
- The bar should be aligned with the content container, right now it's floating and feels out of place
![pic](https://user-images.githubusercontent.com/577802/71139892-8c00ac00-224a-11ea-8354-844d20c95a57.png)


- The bar is very large with reference to other elements on the page and draws a lot of attention (given that it's already a stickied element, we don't need it to be too large), perhaps something smaller will be sufficient.

- On mobile it also floats a little too high above the bottom, it can be aligned with the content container too.

simboonlong committed:
use max-width for @media
add 1 more wrapping div to align floating bar
smaller buttons in floating bar
corrected typo in target="_blank"

simboonlong committed:
remove duplicate css property

simboonlong commented on src/components/CertificateViewer.js:
```diff
@@ -44,57 +44,59 @@ const renderHeaderBlock = (props, childRef) => {
+            <div className="col-auto">
+              <a
+                download={`${props.certificate.id}.opencert`}
+                target="_black"
```
aligned the bar, make the buttons smaller, corrected the typo.

sebastianquek commented on src/styles/variables.scss:
```diff
@@ -75,3 +112,23 @@ $max-width: 1280px;
   }
 }
+
+@mixin media-breakpoint-custom ($min: 0, $max: 0) {
```
I think we should remove this for now, so that the git history makes sense (i.e. we can trace when this was added and used at the same time).

sebastianquek commented on src/components/certificateViewer.scss:
```diff
@@ -130,19 +130,61 @@
+
+  width: 40px;
+  height: 40px;
+  line-height: 40px;
```
Using `line-height` doesn't really center the icon perfectly 😊 , shall we use flex instead?

simboonlong committed:
remove breakpoint mixin
use flex to center instead of line-height

simboonlong committed:
fix: add missing variables

simboonlong commented:
stale, closing
